default/python/AI/PolicyAI.py

```diff
@@ -0,0 +1,235 @@
+import freeOrionAIInterface as fo
+from logging import debug, error
+
+import PlanetUtilsAI
+from aistate_interface import get_aistate
+from EnumsAI import PriorityType
+
+propaganda = "PLC_PROPAGANDA"
+algo_research = "PLC_ALGORITHMIC_RESEARCH"
+moderation = "PLC_MODERATION"
+industrialism = "PLC_INDUSTRIALISM"
+technocracy = "PLC_TECHNOCRACY"
+centralization = "PLC_CENTRALIZATION"
+bureaucracy = "PLC_BUREAUCRACY"
+infra1 = "PLC_PLANETARY_INFRA"
+infra2 = "PLC_SYSTEM_INFRA"
+infra3 = "PLC_INTERSTELLAR_INFRA"
+basics = [
+    propaganda,
+    algo_research,
+    infra1,
+    # no use for the extra slot yet and they are expensive
+    # infra2,
+    # infra3,
+]
+
+
+class PolicyManager:
+    """Policy Manager for one round"""
+
+    def __init__(self):
+        # resourceAvailable includes this turns production, but that is wrong for influence
+        self.empire = fo.getEmpire()
+        self.universe = fo.getUniverse()
+        self.ip = self.empire.resourceAvailable(fo.resourceType.influence) - self.get_infl_prod()
+        self.adopted = set(self.empire.adoptedPolicies)
+        self.adoptable = self.get_adoptable()
+        empire_owned_planet_ids = PlanetUtilsAI.get_owned_planets_by_empire(self.universe.planetIDs)
+        self.planet_ids = PlanetUtilsAI.get_populated_planet_ids(empire_owned_planet_ids)
+        self.num_populated = len(self.planet_ids)
+        self.num_outposts = len(empire_owned_planet_ids) - self.num_populated
+        self.aistate = get_aistate()
+
+    def generate_orders(self):
```

Comment:
This function is very huge. Let's deal with it. 

We could keep it as an index of operation and move all low-level stuff into separate methods. 

```python
def generate_orders():
    debug(...)
    self._adopt_base_policies()
    self._process_bureaucracy()
    ...
```

Reply:
Well, it was quite short compared to other functions in the AI, but it surely can't hurt to split it now, it will surely grow.

Reply:
> Well, it was quite short compared to other functions in the AI

There is a nice metric called [cognitive complexity](https://pypi.org/project/flake8-cognitive-complexity/), it's a measurement of how hard it to read the code.  For my personal (single) project I keep it at 5.   AI has hundreds and thousands.   

So referencing AI code as an example might not be a good idea.